### PR TITLE
Drop 2.6

### LINKED ISSUE #1245: Drop 2.6 and 3.3 support
Should we drop support for Python 2.6 or 3.3?

Historically we have targetted these versions because we intend dask to be an infrastructural library and because these versions are fairly easy to support in the core of the library.

We might consider dropping support for these versions in some submodules (like dask.dataframe) where users are typically more up to date than in other modules (like dask.array) or the core.

### PR DESCRIPTION
**Pull request by mrocklin**
Fixes #1245

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -24,10 +23,7 @@ install:
   # Install dependencies
   - conda create -n test-environment python=$TRAVIS_PYTHON_VERSION
   - source activate test-environment
-  - if [[ $TRAVIS_PYTHON_VERSION == '2.6' ]] || [[ $TRAVIS_PYTHON_VERSION == '3.3' ]]; then conda install numpy=1.9.2 scipy pandas=0.16.2 pytables=3.2.0 h5py=2.5.0 bcolz=0.10.0 pytest coverage toolz scikit-learn cytoolz cloudpickle chest blosc cython psutil ipython; fi
   - if [[ $TRAVIS_PYTHON_VERSION == '2.7' ]] || [[ $TRAVIS_PYTHON_VERSION == '3.4' ]] || [[ $TRAVIS_PYTHON_VERSION == '3.5' ]]; then conda install numpy scipy pandas pytables h5py bcolz pytest coverage toolz scikit-learn cytoolz cloudpickle chest blosc cython psutil ipython distributed; fi
-  - if [[ $TRAVIS_PYTHON_VERSION == '2.6' ]]; then conda install unittest2; fi
-  - if [[ $TRAVIS_PYTHON_VERSION != '2.6' ]]; then conda install bokeh; fi
```
Comment (jcrist) [inline .travis.yml]: Need to add bokeh back in.

File: .travis.yml [hunk 2]
```diff
@@ -24,10 +23,7 @@ install:
   # Install dependencies
   - conda create -n test-environment python=$TRAVIS_PYTHON_VERSION
   - source activate test-environment
-  - if [[ $TRAVIS_PYTHON_VERSION == '2.6' ]] || [[ $TRAVIS_PYTHON_VERSION == '3.3' ]]; then conda install numpy=1.9.2 scipy pandas=0.16.2 pytables=3.2.0 h5py=2.5.0 bcolz=0.10.0 pytest coverage toolz scikit-learn cytoolz cloudpickle chest blosc cython psutil ipython; fi
```
Comment (jcrist) [inline .travis.yml]: There's no branch for 3.3 now.

File: .travis.yml (hunk 1 above)
Comment (mrocklin) [inline .travis.yml]: Fixed

File: .travis.yml (hunk 2 above)
Comment (mrocklin) [inline .travis.yml]: Fixed

**Comment by mrocklin** [comment]
Merging soon if no comments

**Comment by jcrist** [comment]
\o/. Smaller test matrix, and we can finally use dict comprehensions :).

**Pull request merged**